test/rules/avoid_function_literal_in_foreach_method.dart

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N avoid_function_literal_in_foreach_method`
+
+void main() {
+  Iterable<String> people;
+
+  for (var person in people) { // OK
+    print(person);
+  }
+  people.forEach((person) { // LINT
```

Comment:
Given they are concise, I would say no:
```dart
people.forEach((p) => foo(p, 0)); // OK

void foo(String bar, int baz) {}
```

Turning this into a for loop would turn one line into three. Thoughts @bwilkerson